### PR TITLE
.ocamlformat: 0.21 -> 0.24.1

### PR DESCRIPTION
**Pull request by hlian**
This bumps the version in `.ocamlformat` from 0.21 to 0.23. All the other diffs are a result of changes to how ocamlformat works.

Update: bumped it to the latest version, 0.24.1. This was the source of some of the CI failures.

### CODE REVIEW

**Review by Xyene — APPROVED** [review]
LGTM, thanks!

**Comment by Xyene** [comment]
Whoops, looks like this failed CI. I think you may need to edit the workflows.

Separately, after this is merged, could you please send in a follow-up PR adding the resulting commit hash to `.git-blame-ignore-revs`? ([ref](https://akrabat.com/ignoring-revisions-with-git-blame/))

**Comment by hlian** [comment]
> Separately, after this is merged, could you please send in a follow-up PR adding the resulting commit hash to .git-blame-ignore-revs? 

absolutely!

**Comment by hlian** [comment]
@Xyene another workflow approval, please!

**Comment by hlian** [comment]
@Xyene one more workflow approval, please (sorry, forgot to sign commits)

**Comment by hlian** [comment]
@Xyene ready to merge, when you get the chance!

**Comment by Xyene** [comment]
LGTM, thanks!

**Pull request merged**
